presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -518,7 +518,7 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Void> context)
                         .forEach(clause -> map(clause.getRight(), clause.getLeft()));
             }
 
-            return new JoinNode(node.getId(), node.getType(), left, right, canonicalCriteria, canonicalFilter, canonicalLeftHashSymbol, canonicalRightHashSymbol);
+            return new JoinNode(node.getId(), node.getType(), left, right, canonicalCriteria, canonicalizeAndDistinct(node.getOutputSymbols()), canonicalFilter, canonicalLeftHashSymbol, canonicalRightHashSymbol);
```

Comment:
Why would the node below produce `[a, a, b]`? If I remember correctly (and unless something changed along the way), that should never happen except for OutputNode. If anything, the child should produce just `[a, b]`.